medicai/layers/resize.py

```
@@ -0,0 +1,40 @@
+            )
+
+        self.target_shape = target_shape
+        self.interpolation = interpolation
+
+    def build(self, input_shape):
+        self.spatial_dims = len(input_shape) - 2
+        self.channels = input_shape[-1]
+        if self.spatial_dims not in (2, 3):
+            raise ValueError(
+                f"SpatialResize only supports 2D or 3D inputs. Got spatial_dims={self.spatial_dims}"
+            )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)